code/game/machinery/_machinery.dm

```diff
@@ -146,6 +148,18 @@ Class Procs:
 
 	return INITIALIZE_HINT_LATELOAD
 
+/obj/machinery/proc/get_occupant()
+	RETURN_TYPE(/atom/movable)
+	SHOULD_NOT_OVERRIDE(TRUE)
+
+	return occupant
+
+/obj/machinery/proc/set_occupant(atom/movable/new_occupant)
```

Comment:
this is arguably justifiable due to the signal .

Reply:
This is good, setters let us react to events.